src/Raven.Server/Documents/AI/Embeddings/EmbeddingsHelper.cs

```diff
@@ -89,18 +88,18 @@ public static EmbeddingValue CreateEmbeddingValue(ReadOnlyMemory<float> embeddin
                     var span = MemoryMarshal.Cast<float, sbyte>(newMemory.Span);
                     var result = VectorQuantizer.TryToInt8(embedding.Span, span, out usedBytes);
                     Debug.Assert(result, "TryToInt8 should always return true");
-
-                    return new EmbeddingValue(newMemory, usedBytes);
+        
+                    return MemoryMarshalEx.Cast<float, byte>(embedding)[..usedBytes];
```

Comment:
Shouldn't we use `newMemory` here? We want the value _after_ quantization

Reply:
done